### ISSUE TITLE
sitewide 508-defect-2 [SCREENREADER]: Subway list steps SHOULD announce the step number

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9936]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
! Team affected: VSA BAM2; Project: MDOT MVP
```

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The introduction page for various VA applications use a "subway list" pattern. This text generally of the numbers of the subway list **should** read aloud in a screen reader. On this introduction page, the numbers are not read aloud in JAWS+IE11.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to know which step of the process I am on, on par with a sighted user's experience.

## Environment

* Operating System: Windows
* Browser: IE11, Firefox
* Screenreading device: JAWS, NVDA
* Server destination: staging

> Note: VoiceOver+Safari reads the individual steps. I tested another page with the subway steps, Apply for health care benefits, and heard the numbers read by NVDA, but not on this hearing aid batteries intro.

## Steps to Recreate

1. Enter https://staging.va.gov/hearing-aid-batteries-and-accessories/introduction in browser
2. Start screenreading device listed in Environment
3. Begin to read the page
4. Verify that the numbered subway steps are not read aloud, and there is no communication of the progress through the steps for screen reader users.

## Possible Fixes (optional)

See some of the other pages that use the subway list, and use the same code.

## WCAG or Vendor Guidance (optional)

* [wuhcag - consistent identification](https://www.wuhcag.com/consistent-identification/)

* [W3C WCAG - Understanding Success Criterion 3.2.4: Consistent Identification](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html)

**Comment by mr0sari0** [comment]
I suspect this to be a platform related issue or a Firefox related issue with CSS content that's added before or after the element. I tested other forms on Firefox with Pericles and it skips over the number of the step. In the screenshot below, you can see that the names of the steps include the number.

<img width="892" alt="Screen Shot 2020-06-09 at 12 25 59 PM" src="https://user-images.githubusercontent.com/12755283/84174698-ee45a700-aa4c-11ea-816b-19062dc7cbd0.png">

**Comment by jenstrickland** [comment]
@1Copenut — Have you heard of this bug previously? Is this known behavior for the subway list numbers?

I tested `Apply for health care benefits` with Firefox+NVDA and heard each of the numbers read out. However, for this work, the numbers are only read aloud in VoiceOver+Safari.

**Comment by 1Copenut** [comment]
@mr0sari0 In this case, you might have to create code in isolation on a Codepen or other scratch pad, then introduce the CSS rules slowly to see which one breaks the behavior. I'd have thought adding `role="list"` to the OL would have solved it, but I guess that's a UL trick only.

**Comment by jenstrickland** [comment]
@1Copenut  — @mr0sari0 and I hopped on a call and VoiceOver+Safari is reading the numbers. 

Mahariel, let me know if you added the `role="list"` and I can test in JAWS & NVDA.

**Comment by mr0sari0** [comment]
@1Copenut and @jenstrickland, I am actually getting an error for adding `role="list"`. Apparently, OLs have an implicit role of list.

**Comment by jenstrickland** [comment]
Since the list semantics are removed by the CSS `list-style:none` that would remove it, so the error might have to be accepted. 

**However** — this is not an issue on other pages that are using the subway list, and the `role="list"` isn't added. For example: https://www.va.gov/health-care/apply/application/introduction  

I inspected your code in comparison and it looks the same. 

I just tested both urls with JAWS+IE11, and it looks like it's a platform level issue, so I am going to close this issue for your team — as long as the `role="list"` is not added. Would you please confirm that you are not adding it in a PR?


![Screen Shot 2020-06-17 at 1 49 46 PM](https://user-images.githubusercontent.com/57469/84931917-7c4f0c80-b0a1-11ea-804f-0c0ef75be5f8.png)

**Comment by mr0sari0** [comment]
Hi @jenstrickland, thanks for looking deeper into this. I have not added the `role="list"` to the PR.

**Comment by jenstrickland** [comment]
This is a platform level issue, a JAWS+IE11 bug, so I will assign to platform to take a look.

**Comment by 1Copenut** [comment]
Issue moved to [department-of-veterans-affairs/vets-design-system-documentation #247](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/247) via [**ZenHub**](https://www.zenhub.com/)

**Comment by jenstrickland** [comment]
As this bug is still in the VSA BAM2 work, we should keep this bug open. Once the sitewide issue is resolved, we should ensure that the update is reflected in this work.

**Comment by jenstrickland** [comment]
@bradl3yC @leahkeeler This is a sitewide bug, not something that is the responsibility of the BAM2 team. I reassigned to VSP and included the note so they knew where this bug was found. It is not for your team to be concerned with at this time.

![Screen Shot 2020-08-24 at 4 30 12 PM](https://user-images.githubusercontent.com/57469/91093198-1e055380-e627-11ea-9a5d-9f0f67cb33a9.png)

**Comment by bkjohnson** [comment]
This should be fixed in the Web Component implementation.
